tests/ClientTest.php

```diff
@@ -0,0 +1,285 @@
+<?php
```

Comment:
I think this is still missing tests for range headers and chunked downloads of large files.